### PR TITLE
Proposals step 1: clean up invites

### PR DESCRIPTION
**Pull request by hubudibu**
- invitation step in group creation, invitation later and member proposal uses same view
- invitation and proposal views had ~10 lines of significant difference, so they moved to the same file now
- removed unnecessary extra step from responding to invite
- tried to improve readability

### CODE REVIEW

File: frontend/simple/js/invites.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  // create record of the invitation in the group's contract
+  return new Events.HashableGroupRecordInvitation(
+    {
+      username: memberName,
+      inviteHash: inviteHash,
+      sentDate
+    },
+    latest
+  )
+}
```
Comment (taoeffect) [inline frontend/simple/js/invites.js:33]: FYI, these kinds of functions are a good candidate for an SBP namespace (along with a bunch of other similar functions). Can you put a TODO comment at the top to convert to SBP?

File: frontend/simple/views/Invite.vue [hunk 2]
```diff
@@ -86,105 +33,109 @@
...
-              headers: [this.$store.state.currentGroupId],
-              messageType: Events.HashableMailboxPostMessage.TypeInvite,
-              sentDate
-            },
-            mailbox
-          )
-          await backend.publishLogEntry(member.state.attributes.mailbox, invite)
+          if (this.isAlreadyInvited(member)) {
+            template(L('Failed to invite users: member {name} is already invited'),
+              { name: memberName })
```
Comment (taoeffect) [inline frontend/simple/views/Invite.vue]: This line doesn't appear to be doing anything.

File: frontend/simple/views/Invite.vue [hunk 3]
```diff
@@ -86,105 +33,109 @@
...
+               { contractId: mailbox, action: JSON.stringify(inviteToMailbox.toObject()) },
+               { contractId: groupId, action: JSON.stringify(inviteToGroup.toObject()) }
+              ],
+              initiator: this.loggedIn.name,
+              initiationDate: new Date().toString()
+            }, latest)
+            await backend.publishLogEntry(groupId, proposal)
+          } else {
+            await backend.publishLogEntry(mailbox, inviteToMailbox)
+            await backend.publishLogEntry(groupId, inviteToGroup)
```
Comment (taoeffect) [inline frontend/simple/views/Invite.vue:123]: So we need to hop on a call to talk about this sort of situation, where line 122 succeeds but line 123 fails. This is the kind of scenario that Victor's other PR was about handling.

File: frontend/simple/views/ProposeMember.vue [hunk 4]
```diff
@@ -1,165 +0,0 @@
...
-        // TODO: Create More descriptive errors
-        this.userErrorMsg = L('Failed to Propose Users')
-      }
-    },
-    remove (index) {
-      this.members.splice(index, 1)
-    }
-  }
-}
-</script>
```
Comment (taoeffect) [inline frontend/simple/views/ProposeMember.vue:165]: What happened to this file and why?

File: frontend/simple/js/invites.js (hunk 1 above)
Comment (hubudibu) [reply frontend/simple/js/invites.js:33]: sure! I'll look into this, seems like a good candidate to try my hand with SBP :)

File: frontend/simple/views/Invite.vue (hunk 2 above)
Comment (hubudibu) [reply frontend/simple/views/Invite.vue]: 🙈 yeah I just forgot the first half

File: frontend/simple/views/Invite.vue (hunk 3 above)
Comment (hubudibu) [reply frontend/simple/views/Invite.vue:123]: sounds interesting! let's talk!

File: frontend/simple/views/ProposeMember.vue (hunk 4 above)
Comment (hubudibu) [reply frontend/simple/views/ProposeMember.vue:165]: "invitation and proposal views had ~10 lines of significant difference, so they moved to the same file now" => this is also handled by Invite.vue now

**Comment by taoeffect** [comment]
OK, since this PR is a subset of #355 (which contains everything from this and continues onward from the review here) I'm closing this PR and continuing my review there.

**Pull request closed without merging**